### PR TITLE
Autodoc confdefs

### PR DESCRIPTION
**Pull request by vEpiphyte**
- Part out devops guide into discrete portions
- Update autodoc to generate a enumerated list of configuration options for a Cell
- UPdate autodoc to be able to generate RST documents for stormservice configurations.
- Add configuration notes about the Cell.
- Add autodoc data to the docs for the Cortex, Axon, Cryotank and SyncMigrator

### CODE REVIEW

File: synapse/tools/autodoc.py [hunk 1]
```diff
@@ -347,6 +349,81 @@ async def docModel(outp,
...
+        rst.addHead(hname, lvl=1)
+
+        desc = conf.get('description', nodesc)
+        if not desc.endswith('.'):  # pragma: no cover
+            logger.warning(f'Description for [{name}] is missing a period.')
+
+        lines = []
+        lines.append(desc)
+
+        extended_description = conf.get('extended_description')
```
Comment (vEpiphyte) [inline synapse/tools/autodoc.py:393]: This allows us to embed reform sphinx RST into confdefs

**Comment by vEpiphyte** [comment]
Closing in favor if #1771 since this branch got merged there.

File: docs/synapse/devguides/devops_cell.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+Config File
+***********
+
+A Cell has one optional configuration file, ``cell.yaml``, that may be located in the root Cell directory.
+The format of this file is YAML, and variable names are specified without alteration, for example::
+
+    ---
+    auth:passwd: secret
+    mysvc:apikey: 720a50f9-cfa1-43a9-9eca-dda379ecd8c5
+    ---
```
Comment (Cisphyx) [inline docs/synapse/devguides/devops_cell.rst:38]: I think the end of this block should be `...` not `---`

File: docs/synapse/devguides/devops_cell.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+Environment Variables
+*********************
+
+Environment variable names are automatically generated for a Cell service using the following naming convention:
+``SYN_<cell_subclass_name>_<variable_name>``.  Variable names with colons are replaced with underscores,
+and the raw environmental variable value is deserialized as yaml, prior to performing type validation.
```
Comment (Cisphyx) [inline docs/synapse/devguides/devops_cell.rst:45]: Was 'environmental' variable intended or a typo?

File: docs/synapse/devguides/devops_cell.rst (hunk 2 above)
Comment (vEpiphyte) [reply docs/synapse/devguides/devops_cell.rst:38]: fixed

File: docs/synapse/devguides/devops_cell.rst (hunk 3 above)
Comment (vEpiphyte) [reply docs/synapse/devguides/devops_cell.rst:45]: typo

File: docs/synapse/devguides/devops_cell.rst [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+.. note::
+
+    When used to launch a Cell as the entry point of a program, the configuration precedence order is:
+
+    #. Command line arguments
+    #. Environment variables
+    #. cell.yaml values
+
+    These may all be mixed and matched for a given deployment.
+    If a backup of a service is made and the deployment uses configuration data from command line arguments and environmental variables,
```
Comment (Cisphyx) [inline docs/synapse/devguides/devops_cell.rst:62]: another 'environmental'

Comment (vEpiphyte) [reply docs/synapse/devguides/devops_cell.rst]: typo

**Pull request merged**